File: MovieTicketBooking/Movie Booking C#/Model/SeatLock.cs
namespace MovieTicketBooking.Model
{
    public class SeatLock
    {
        public string LockedBy { get; }
        public Seat Seat;
        public Show Show;
        int TimeoutInSeconds;
        DateTime LockTime;
        public SeatLock(string lockedBy, Seat seat, Show show, int timeoutInSeconds, DateTime lockTime)
        {
            this.LockedBy = lockedBy;
            this.Seat = seat;
            this.Show = show;
            this.TimeoutInSeconds = timeoutInSeconds;
            this.LockTime = lockTime;
        }
        public bool isLockExpired()
        {
            DateTime lockInstant = LockTime.ToUniversalTime().AddSeconds(TimeoutInSeconds);
            DateTime currentInstant = DateTime.UtcNow;
            return lockInstant < currentInstant;
        }
    }
}

File: MovieTicketBooking/Movie Booking Java/src/main/java/org/example/model/SeatLock.java
package org.example.model;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.Duration;
import java.time.LocalDateTime;

@AllArgsConstructor
@Getter
public class SeatLock {

    private final Seat seat;
    private final Show show;
    private final LocalDateTime lockTime;
    private final Integer durationInMinutes;
    private final String lockedBy;

    public boolean isLockExpired() {

        LocalDateTime lockEndTime = lockTime.plus(Duration.ofMinutes(durationInMinutes));
        return LocalDateTime.now().isAfter(lockEndTime);
    }
}
